crates/viewer/re_renderer/shader/conversions/chroma_subsampling_converter.wgsl

```diff
@@ -0,0 +1,112 @@
+#import <../types.wgsl>
+#import <../screen_triangle_vertex.wgsl>
+
+struct UniformBuffer {
+    format: u32,
+    primaries: u32,
+    target_texture_size: vec2u,
+};
+
+@group(0) @binding(0)
+var<uniform> uniform_buffer: UniformBuffer;
+
+@group(0) @binding(1)
+var input_texture: texture_2d<u32>;
+
+
+const FORMAT_Y_UV = 0u;
+const FORMAT_YUYV16 = 1u;
+
+const PRIMARIES_BT601 = 0u;
+const PRIMARIES_BT709 = 1u;
+
+
+/// Returns sRGB from YUV color.
+///
+/// This conversion mirrors the function in `crates/store/re_types/src/datatypes/tensor_data_ext.rs`
+///
+/// Specifying the color standard should be exposed in the future [#3541](https://github.com/rerun-io/rerun/pull/3541)
+fn srgb_from_yuv(yuv: vec3f, primaries: u32) -> vec3f {
+    // rescale YUV values
+    let y = (yuv[0] - 16.0) / 219.0;
+    let u = (yuv[1] - 128.0) / 224.0;
+    let v = (yuv[2] - 128.0) / 224.0;
+
+    var rgb: vec3f;
+
+    switch (primaries) {
+        // BT.601 (aka. SDTV, aka. Rec.601). wiki: https://en.wikipedia.org/wiki/YCbCr#ITU-R_BT.601_conversion
+        // Also note according to https://en.wikipedia.org/wiki/SRGB#sYCC_extended-gamut_transformation
+        // > Although the RGB color primaries are based on BT.709,
+        // > the equations for transformation from sRGB to sYCC and vice versa are based on BT.601.
+        case PRIMARIES_BT601: {
+            rgb.r = y + 1.402 * v;
+            rgb.g = y - 0.344 * u - 0.714 * v;
+            rgb.b = y + 1.772 * u;
+        }
+
+        // BT.709 (aka. HDTV, aka. Rec.709). wiki: https://en.wikipedia.org/wiki/YCbCr#ITU-R_BT.709_conversion
+        case PRIMARIES_BT709: {
+            rgb.r = y + 1.575 * v;
+            rgb.g = y - 0.187 * u - 0.468 * v;
+            rgb.b = y + 1.856 * u;
+        }
+
+        default: {
+            rgb = vec3f(1.0, 0.0, 1.0); // Magenta to indicate trouble
```

Comment:
I think we have an ERROR_COLOR in one of the .wgsl files. If not, we should add it.

Reply:
ohhh you're right, forgot about it!